contrib/boot-contracts-stateful-prop-tests/tests/pox-4/err_Commands.ts

```diff
@@ -0,0 +1,2283 @@
+          r.operator,
+          r.increaseBy,
+          r.authId,
+          function (
+            this: StackIncreaseAuthCommand_Err,
+            model: Readonly<Stub>,
+          ): boolean {
+            const stacker = model.stackers.get(this.wallet.stxAddress)!;
+            if (
+              model.stackingMinimum > 0 &&
+              stacker.isStacking &&
+              stacker.isStackingSolo &&
+              !stacker.hasDelegated &&
+              stacker.amountLocked > 0 &&
+              !(this.increaseBy <= stacker.amountUnlocked) &&
+              this.increaseBy >= 1
+            ) {
+              model.trackCommandRun(
+                "StackIncreaseAuthCommand_Err_Stacking_Insufficient_Funds",
+              );
+              return true;
+            } else return false;
+          },
+          POX_4_ERRORS.ERR_STACKING_INSUFFICIENT_FUNDS,
+        ),
+    ),
+    // StackIncreaseAuthCommand_Err_Stacking_Invalid_Amount
+    fc.record({
+      operator: fc.constantFrom(...wallets.values()),
+      increaseBy: fc.constant(0),
+      authId: fc.nat(),
+    }).map(
+      (r) =>
+        new StackIncreaseAuthCommand_Err(
+          r.operator,
+          r.increaseBy,
+          r.authId,
+          function (
+            this: StackIncreaseAuthCommand_Err,
+            model: Readonly<Stub>,
+          ): boolean {
+            const stacker = model.stackers.get(this.wallet.stxAddress)!;
+            if (
+              model.stackingMinimum > 0 &&
+              stacker.isStacking &&
+              stacker.isStackingSolo &&
+              !stacker.hasDelegated &&
+              stacker.amountLocked > 0 &&
+              this.increaseBy <= stacker.amountUnlocked &&
+              !(this.increaseBy >= 1)
+            ) {
+              model.trackCommandRun(
+                "StackIncreaseAuthCommand_Err_Stacking_Invalid_Amount",
+              );
+              return true;
+            } else return false;
+          },
+          POX_4_ERRORS.ERR_STACKING_INVALID_AMOUNT,
+        ),
+    ),
+    // StackExtendSigCommand_Err_Stacking_Is_Delegated_1
+    fc.record({
+      wallet: fc.constantFrom(...wallets.values()),
+      authId: fc.nat(),
+      extendCount: fc.integer({ min: 1, max: 12 }),
+      currentCycle: fc.constant(currentCycle(network)),
+    }).map(
+      (r: {
+        wallet: Wallet;
+        extendCount: number;
+        authId: number;
+        currentCycle: number;
+      }) =>
+        new StackExtendSigCommand_Err(
+          r.wallet,
+          r.extendCount,
+          r.authId,
+          r.currentCycle,
+          function (
+            this: StackExtendSigCommand_Err,
+            model: Readonly<Stub>,
+          ): boolean {
+            const stacker = model.stackers.get(this.wallet.stxAddress)!;
+
+            const firstRewardCycle =
+              stacker.firstLockedRewardCycle < this.currentCycle
+                ? this.currentCycle
+                : stacker.firstLockedRewardCycle;
```

Comment:
```suggestion
            const firstRewardCycle = Math.max(stacker.firstLockedRewardCycle, this.currentCycle);
```

This pattern is used multiple times in this file (I'm just commenting here)

Reply:
Thank you! I have replaced the way `firstRewardCycle` is calculated all over the stateful property testing environment.